xbmc/addons/settings/AddonSettings.cpp

```diff
@@ -683,8 +676,6 @@ std::shared_ptr<CSettingCategory> CAddonSettings::ParseOldCategoryElement(uint32
 
 bool CAddonSettings::InitializeFromOldSettingDefinitions(const CXBMCTinyXML& doc)
 {
-  m_logger->debug("trying to load setting definitions from old format...");
```

Comment:
Maybe keep this one and encourage add-on developers to migrate to the new settings format? If we remove this on v20 we should probably advise in v19

Reply:
Is there anywhere the warning could go that only shows once per addon call? Currently it shows every time a setting is read.

Reply:
It should only show when loading the settings for an add-on. It shouldn't show on every access to the setting from the same script. Otherwise something else is wrong. But I agree that it is logged out too often especially for non-service add-ons. But for debugging setting issues it's still important information to know that Kodi tries to load old style settings.

So if we can find a way to show it once for every add-on with settings that's ok with me.

Reply:
Ill double check when it's getting logged.

Could leave in 19 and remove from 18 (if back port)?

Reply:
> Maybe keep this one and encourage add-on developers to migrate to the new settings format?

are there any guidelines?
the last time i made a conversion (last year) attempt i noticed a lack of ability to run "actions" with python add-on

Reply:
I see this logged all the time and my addons use the new setting format. Does it default to checking the old format first?

Reply:
For add-ons using the old settings style logging should look like this:
```
loading setting definitions
trying to load setting definitions from old format...
```
For add-ons using the new settings style logging should look like this:
```
loading setting definitions
```

Reply:
Apologies, you are correct. I had mixed up Leia (old settings) vs Matrix (new settings).